java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -274,11 +314,9 @@ public ArrowBuf buffer(final int initialRequestSize, BufferManager manager) {
       return empty;
     }
 
-    // round to next largest power of two if we're within a chunk since that is how our allocator
+    // round to a power of two if we're within a chunk since that is how our allocator
```

Comment:
could you expand on this comment a little bit. It seems strange to return a value less then the requested amount of memory.

Reply:
Sure. Thanks for your kind reminder.

Reply:
> Sure. Thanks for your kind reminder. Please see if it is what you want.